### ISSUE TITLE
reckon can't learn from a file with the "wrong" date format

### ISSUE THREAD

**Issue by TheLugal** [cantino/reckon#130]
Hello, thank you for maintaining this great program!

I have the following `sample.journal` file. It is generated by reckon itself using the flag `--ledger-date-format %Y.%m.%d`.

```ledger
2024.03.12	groceries; 11223344556; 32095205940
	assets:bank:spending			 530.00 NOK
	assets:bank:co:groceries			

2024.03.12	autosave; 11223344555; 11223344556
	assets:bank:savings			
	assets:bank:spending			-10.00 NOK
```

When I try to set reckon to learn from this, I get an info message about `Unknown entry type`. If I change the date to `%Y-%m-%d` in the sample, it behaves as expected.

I use the command `reckon -v -f demo.csv -l sample.journal -o output.journal`
```log
What is this account named in Ledger  |Assets:Bank:Checking|?
assets:bank:spending
I, [2024-03-26T19:18:15.506110 #2754156]  INFO -- : suggested file encoding: utf-8
I, [2024-03-26T19:18:15.506184 #2754156]  INFO -- : guessing ; as csv separator
Using column 3 as the money column.  Use --money-colum to specify a different one.
I, [2024-03-26T19:18:15.509045 #2754156]  INFO -- : learning from #<File:0x00006060e2eb54d0>
I, [2024-03-26T19:18:15.509091 #2754156]  INFO -- : Unknown entry type: 2024.03.12      groceries; 11223344556; 32095205940
I, [2024-03-26T19:18:15.509109 #2754156]  INFO -- : Unknown entry type:         assets:bank:spending                     530.00 NOK
I, [2024-03-26T19:18:15.509122 #2754156]  INFO -- : Unknown entry type:         assets:bank:co:groceries
I, [2024-03-26T19:18:15.509134 #2754156]  INFO -- : Unknown entry type:
I, [2024-03-26T19:18:15.509147 #2754156]  INFO -- : Unknown entry type: 2024.03.12      autosave; 11223344555; 11223344556
I, [2024-03-26T19:18:15.509159 #2754156]  INFO -- : Unknown entry type:         assets:bank:savings
I, [2024-03-26T19:18:15.509171 #2754156]  INFO -- : Unknown entry type:         assets:bank:spending                    -10.00 NOK
```

I suggest reckon make use of `--ledger-date-format` in the learning file by default if present, and have a `--learning-date-format` flag for customizability.

**Comment by benprew** [comment]
Hi @TheLugal thanks for using Reckon! I just pushed a fix (0.9.6). Can you update reckon and give it a try?

Thanks!

**Comment by TheLugal** [comment]
Thank you!

It seems to be working swimmingly now :)
